src/test/java/bio/overture/ego/controller/CreateUserControllerTest.java

```diff
@@ -0,0 +1,761 @@
+    val newUser =
+        initStringRequest()
+            .endpoint("/users/%s", tokenInfo.getId())
+            .getAnd()
+            .assertOk()
+            .extractOneEntity(User.class);
+
+    // assert next user matches idToken
+    assertEquals(newUser.getProviderType(), idToken.getProviderType());
+    assertEquals(newUser.getProviderId(), idToken.getProviderId());
+    assertEquals(newUser.getEmail(), idToken.getEmail());
+    assertEquals(newUser.getFirstName(), idToken.getGivenName());
+    assertEquals(newUser.getLastName(), idToken.getFamilyName());
+
+    val existingUser =
+        initStringRequest()
+            .endpoint("/users/%s", migratedUser.getId())
+            .getAnd()
+            .assertOk()
+            .extractOneEntity(User.class);
+
+    // assert user1 and user2 are distinct users with same email
+    assertNotEquals(newUser.getId(), existingUser.getId());
+    assertNotEquals(newUser.getProviderType(), existingUser.getProviderType());
+    assertNotEquals(newUser.getProviderId(), existingUser.getProviderId());
+    assertEquals(newUser.getEmail(), existingUser.getEmail());
+
+    // assert migratedUser has not updated providerId
+    assertEquals(migratedUser.getProviderId(), migratedUser.getEmail());
+    assertEquals(migratedUser.getProviderId(), existingUser.getProviderId());
+  }
+
+  @Test
+  public void idTokenHasProviderInfoButNoEmail_createUser() {
+    idToken.setProviderType(GOOGLE);
+    idToken.setProviderId(generateNonExistentProviderId(userService));
+    idToken.setGivenName("UserHas");
+    idToken.setFamilyName("NoEmail");
+
+    val response =
+        initStringRequest()
+            .endpoint("/oauth/google/token")
+            .headers(tokenHeaders)
+            .getAnd()
+            .assertOk()
+            .getResponse()
+            .getBody();
+
+    // assert valid token is returned
+    assertTrue(tokenService.isValidToken(response));
+
+    val user = tokenService.getTokenUserInfo(response);
+    // user is created without an email
+    val newUser =
+        initStringRequest()
+            .endpoint("/users/%s", user.getId())
+            .getAnd()
+            .assertOk()
+            .extractOneEntity(User.class);
+
+    // assert new user matches idToken
+    assertEquals(newUser.getProviderType(), idToken.getProviderType());
+    assertEquals(newUser.getProviderId(), idToken.getProviderId());
+    assertEquals(newUser.getEmail(), idToken.getEmail());
+    assertEquals(newUser.getFirstName(), idToken.getGivenName());
+    assertEquals(newUser.getLastName(), idToken.getFamilyName());
+
+    // assert email is null
+    assertTrue(isNull(newUser.getEmail()));
+  }
+
+  @Test
+  public void createUser_BlankProviderId_BadRequest() {
+    val idToken = entityGenerator.setupUserIDToken(DEFAULT_PROVIDER_TYPE, "");
+
+    exceptionRule.expect(MalformedRequestException.class);
+    exceptionRule.expectMessage("Provider id cannot be blank.");
+    userService.getUserByToken(idToken);
```

Comment:
Leave a comment saying we dont need to test providerType being null, since we hardcode it to not be null via, SSOFilters